packages/lib/slots.ts

```diff
@@ -192,18 +195,46 @@ function buildSlotsWithDateRanges({
       ? range.start
       : startTimeWithMinNotice;
 
-    slotStartTime =
-      slotStartTime.minute() % interval !== 0
-        ? slotStartTime.startOf("hour").add(Math.ceil(slotStartTime.minute() / interval) * interval, "minute")
-        : slotStartTime;
-
     // Adding 1 minute to date ranges that end at midnight to ensure that the last slot is included
     const rangeEnd = range.end
       .add(dayjs().tz(organizerTimeZone).utcOffset(), "minutes")
       .isSame(range.end.endOf("day").add(dayjs().tz(organizerTimeZone).utcOffset(), "minutes"), "minute")
       ? range.end.add(1, "minute")
       : range.end;
 
+    if (slotStartTime.minute() % interval !== 0) {
+      if (showOptimizedSlots) {
+        // if showOptimizedSlots option is selected, the slotStartTime should not be modified,
+        // so that maximum possible slots are shown.
+        // The below logic in this entire `if branch` only tries to add an increment if sufficient minutes are available (after max possible slots are consumed),
+        // so that slots are shown respecting the 'Start of the Hour'.
+        const startTimeMinutes = slotStartTime.minute() % interval;
+        const minutesRequiredToMoveToNextSlot = interval - startTimeMinutes;
+        const extraMinutesAvailable = rangeEnd.diff(slotStartTime, "minutes") % interval;
+
+        if (extraMinutesAvailable >= minutesRequiredToMoveToNextSlot) {
+          // For cases like, Availability -> 9:05 - 12:00, 60Min EventTypes.
+          // Total available minutes are 175, so only 2 60Min slots can be provided max
+          // And stll 175-120 = 55mins are available, hence 'slotStartTime' is pushed to 10:00 to respect 'Start of the Hour'.
+          // Slots will be shown as '10:00, 11:00' instead of '09:05, 10:05'
+          slotStartTime = slotStartTime.add(minutesRequiredToMoveToNextSlot, "minute");
+        } else if (extraMinutesAvailable >= minutesRequiredToMoveToNextSlot - Math.ceil(interval / 2)) {
```

Comment:
I am unsure about this part. Maybe we should just always use the next 15-minute slot if that still allows us to fit the same amount of slots. 

So in your example the slotStarTime would be 9:15 instead of 9:30

cc @emrysal

Reply:
yes, with check in place for minutes available to move after consuming max slots, we can always move to next 15min slot (if moving to next interval slot is not possible).

Can you please confirm w.r.t example show in the issue description.
For example 1 shown below, the new expected slots would be :
8:00 - 9:00
9:00 - 10:00
_10:15 - 11:15
11:15 - 12: 15_
<img width="837" alt="Screenshot 2024-09-24 at 1 21 17 PM" src="https://github.com/user-attachments/assets/30afc646-e72c-4cf6-97ac-a30cb9412984">

Example 2 - expected slots would still be the same.

Reply:
I changed example 1, as you said the slot should be 10:15-11:15. We should always move to the next 15-minute slot if possible, if that would cause us to lose a slot, then we can start as early as possible even if it's no 15-minute slot

Reply:
sure will update accordingly, Thank you! for confirming and updating.

Reply:
updated and tested